blog/_posts/2024-08-19-KAITO-finetuning.md

```diff
@@ -0,0 +1,104 @@
+
+To use your new fine-tuned adapter, the inference workspace can now accept one or more adapters whose strength values can be adjusted (default of 1.0) to customize model behavior. 
+
+Here is an example of what a fine-tuned inference workspace looks like:
+
+```bash
+apiVersion: kaito.sh/v1alpha1
+kind: Workspace
+metadata:
+  name: workspace-phi-3-mini-adapter
+resource:
+  instanceType: Standard_NC6s_v3
+  labelSelector:
+    matchLabels:
+      apps: phi-3-adapter
+inference:
+  preset:
+    name: phi-3-mini-128k-instruct
+  adapters:
+    - source:
+        name: myadapter
+        image: myregistry.azurecr.io/adapters/myadapter:0.0.1
+        imagePullSecrets:
+          - myregistrysecret
+      strength: "1.0"
+```
+
+As new inference requests come in, they'll flow through the model and adapters so the response will be altered based on new training data and adjusted based on the strength of each adapter. 
+
+Here is an example of what an un-tuned `phi-3-mini-128k-instruct` model response looks like, versus a fine-tuned `phi-3-mini-128k-instruct` model:
+
+![Screenshot of untuned model response](/blog/assets/images/kaito_untuned_phi3_response.png)
+*Screenshot of un-tuned model response*
+
+![Screenshot of tuned model response](/blog/assets/images/kaito_tuned_phi3_response.png)
+*Screenshot of tuned model response*
+
+You can see in the two screenshots above, that the base `phi-3-mini-128k-instruct` model with no fine-tuning wasn’t quite sure of what the user means by “AKS”. However, with a little bit of fine-tuning with a [dataset](https://huggingface.co/datasets/ishaansehgal99/kubernetes-reformatted-remove-outliers) optimized for Kubernetes and cloud platforms, it is quickly able to discern that the user is looking for an answer related to Kubernetes and provides a better response. There is still a bit of work to do when it comes to prompt engineering and/or fine-tuning tweaks but it’s clearly on the right path!
+
+**Summary**
+
+With KAITO’s new fine-tuning capability, you can now give open-source models additional training for specific tasks, which will generate more context-aware results to your business needs. KAITO supports small language model (SLM) families like Phi-3, which you can choose to reduce costs and fine-tune quickly in comparison to larger, general-purpose models. The best part is that as new data flows in, you can iterate with multiple adapters to improve model performance and achieve this sort of ML workflow all within your AKS cluster and with just a few lines of YAML!
```

Comment:
```suggestion
Users expect LLMs to stay on task and provide accurate answers. While foundation models on their own have a wealth of knowledge, they often need to be refined for specific tasks. This can be achieved by leveraging techniques like fine-tuning or RAG, but the process of implementing these techniques is not always straightforward. As you saw, with KAITO, it becomes **really easy**. All you need to do is specify an input dataset, a GPU SKU, and off it goes! KAITO will take care of the rest. Its adapter design makes it very portable to attach to multiple workspaces or even attach multiple adapters to a single workspace. In the end, we can see that the inference and tuning workspaces along with adapters can become fundamental building blocks in your overall ML workflow.
```